### DISCUSSION TITLE
Reverse of Loop

### DISCUSSION THREAD

**Discussion by ghost** [mecha-cms/mecha#98] [Q&A] [answered]
Selamat pagi di akhir pekan minggu pertama,

Hai Mas Taufik,

Bagaimana tekniknya membalik postingan, maksud saya, memunculkan postingan paling lama terlebih dahulu ketimbang yang terbaru, seperti di Blogger kita bisa menggunakan fungsi `reverse` untuk menampilkan postingan terkawak sebagai yang teratas di pages.

**Comment by taufik-nurrohman** [comment, marked as answer]
Ekstensi _page_ memungkinkan untuk mengubah logika penyortiran berdasarkan nilai bidang `sort` pada berkas laman induk. Caranya seperti ini:

~~~ .txt
.\
└── lot\
    └── page\
        ├── article\
        │   ├── article-1.page
        │   ├── article-2.page
        │   └── article-3.page
        ├── article.page
        └── index.page
~~~

Pastikan masing-masing berkas `article-1.page`, `article-2.page` dan `article-3.page` memiliki bidang `time` dengan nilai berupa tanggal sebagai `YYYY-MM-DD hh:mm:ss`. Kalau membuat laman memakai ekstensi [panel](https://github.com/mecha-cms/x.panel), bidang `time` akan dibuat secara otomatis sebagai berkas terpisah.

~~~ .txt
.\
└── lot\
    └── page\
        ├── article\
        │   ├── article-1.page
        │   │   └── time.data
        │   ├── article-2.page
        │   │   └── time.data
        │   └── article-3.page
        │       └── time.data
        ├── article.page
        └── index.page
~~~

Kalau tidak ada bidang `time`, maka ekstensi akan menganggap `filectime($page->path)` sebagai tanggal pembuatan laman.

Setelah itu pada berkas `.\lot\page\article.page`, tambahkan bidang `sort` dengan nilai berupa `[-1, time]` seperti ini:

~~~ .yaml
---
title: Article
description: Lorem ipsum dolor sit amet.
type: Markdown
sort: [-1, time]
...

The article page.
~~~

`-1` untuk urutan mundur, `1` untuk urutan maju. `time` adalah bidang yang akan dijadikan sebagai referensi penyortiran.

Kalau mau menyortir laman berdasarkan judul, maka nilainya harus diganti menjadi `[1, title]`.

**Contoh:** https://github.com/mecha-cms/mecha/blob/c825d9fe5be913f5be2068046ff89c9a20da914c/lot/page/article.page#L6

**Reply by ghost** [reply]
_Matur suwun_ ilmunya, Mas. Saya baru belajar pakai **Mecha**, dan masih menjalankannya secara _offline_.
